### DISCUSSION TITLE
Comma instead of dot with decimal numbers at SpinBox

### DISCUSSION THREAD

**Discussion by AsimovMo** [israel-dryer/ttkbootstrap#328] [Q&A] [answered]
Hello!
I am getting comma instead of dot with decimal numbers at SpinBox. Works just fine when using ttk original library. Switch to ttkbootstrap and i get this error, which generates inconsistency with mathematical functions and manipulations...

Attached some prints:

1 - Code:
![bugSPIN](https://user-images.githubusercontent.com/110856073/183528673-30858e44-e2a8-4004-927e-877e2ec4f401.png)

2 - Output, getting comma in the decimal numbers with ttkbootstrap:
![bugSPIN_Display](https://user-images.githubusercontent.com/110856073/183526961-232c3044-33e7-478c-bfa9-4c6fd674f57b.png)

3 - The same code, getting dot in the decimal numbers with ttk original library:
![bugSPIN_Display2](https://user-images.githubusercontent.com/110856073/183527111-565c97fa-fb22-4d4e-9ca7-073ebd1078f5.png)

I noticed this happen while using the increment or decrement within the SpinBox. Maybe something about the increment attribute?
Even using textvariable to define initial default values with dots it turns to commas if i try to increment/decrement.

Can anyone test this and tell it's a bug or if there is any turnaround?

Thanks in advance!

**Comment by antrrax** [comment, marked as answer]
Isso não deve ser um bug.
Provavelmente seu sistema operacional está em portugues e o ttkbootstrap usa as variaveis de ambiente do idioma do seu sistema operacional. O ttk original usa as variáveis em ingles.

Em portugues o separador decimal é a **virgula** e não o ponto. Basta voce adequar seu código a interface gráfica.

veja no arquivo widgets.py essa linha 13:
`import locale`

---------

para referencias futuras:
https://phrase.com/blog/posts/beginners-guide-to-locale-in-python/
https://herrmann.tech/pt/blog/2021/02/05/como-tratar-dados-com-formatacao-internacional-no-python.html

```
locale.atof (convert a string to a floating point number)
locale.atoi (convert a string to integer)
```

**Reply by AsimovMo** [reply]
Muito obrigado, antrrax.
Realmente o problema foi ocasionado pela tradução de inglês para português.
Pesquisei e estudei um pouco a respeito da função **locale** e consegui resolver o problema.
